core/src/AbilityHelpers.ts

```diff
@@ -170,7 +170,14 @@ class AbilityHelpers implements Types.AbilityHelpersCore, Types.AbilityHelpersIn
 }
 
 export function createAbilityHelpers(win: Window, props?: Types.AbilityHelpersCoreProps): Types.AbilityHelpersCore {
-    return new AbilityHelpers(win, props);
+    if (abilityHelpersExists(win)) {
+        throw new Error(' An Ability helpers instance already exists on this window, you might use `abilityHelpersExists` to validate creation');
+    }
+
+    const ah = new AbilityHelpers(win, props);
+    // @ts-ignore
```

Comment:
Let's not use ts-ignore. Also all internal things ability-helpers stores on the elements are prefixed __ah.
Look at the WindowWithUID example in Utils.ts.
Let's define WindowWithAbilityHelpersInstance with __ahInstance prop. Makes sense to not make it exported, but to make internally `(win as WindowWithAbilityHelpersInstance).__ahInstance = ah;`. That way it's less error-prone to the future refactorings.

Reply:
Added an internal (non-exported) interface